### DISCUSSION TITLE
How do I catch and handle this error myself?

### DISCUSSION THREAD

**Discussion by hashfi-motion** [telegraf/telegraf#1954] [Q&A] [answered]
How do I catch error `Bad Request: query is too old and response timeout expired or query ID is invalid` so I can handle it myself? I already used `bot.catch(....);` but that seems not to work.

Here are the error in more detail:
```error
throw new error_1.default(data, { method, payload });
                  ^
TelegramError: 400: Bad Request: query is too old and response timeout expired or query ID is invalid
    at Telegram.callApi (E:\.......\node_modules\telegraf\lib\core\network\client.js:302:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  response: {
    ok: false,
    error_code: 400,
    description: 'Bad Request: query is too old and response timeout expired or query ID is invalid'
  },
  on: {
    method: 'answerCallbackQuery',
    payload: { text: undefined, callback_query_id: 'xxxxxxxxxx' }
  }
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Note: I haven't tried to wrap my `bot.command(...); bot.action(...); ....etc.` code with `try...catch`, but does using it work? I can't confirm it since I can't replicate the error anymore, if you know how to replicate the error please let me know

**Comment by MKRhere** [comment, marked as answer]
You must await your API calls, for example `await ctx.answerCbQuery(...)`, otherwise Telegraf cannot handle errors.

**Reply by hashfi-motion** [reply]
Huh I guess that's why, since there's couple API call that I don't use `await`. I'll mark this as the answer for now but I haven't got this error again to confirm it fix my issue.

Edit: Can confirm, this does fix my issue

**Reply by Flmob** [reply]
Tried your solution. Testing now. Let's see how it will work. At least it shouldn't make things worse
